app/lib/rocketchat.js

```diff
@@ -254,6 +254,41 @@ const RocketChat = {
 
 			this.usersListener = this.sdk.onStreamData('users', protectedFunction(ddpMessage => RocketChat._setUser(ddpMessage)));
 
+			this.rolesListener = this.sdk.onStreamData('stream-roles', protectedFunction(async(ddpMessage) => {
+				const { type, _id, description } = ddpMessage.fields.args[0];
+				if (/changed/.test(type)) {
+					const db = database.active;
+					const rolesCollection = db.get('roles');
+					try {
+						const rolesRecord = await rolesCollection.find(_id);
+						await db.action(async() => {
+							await rolesRecord.update((u) => {
+								u.description = description;
+							});
+						});
+					} catch (err) {
+						await db.action(async() => {
+							await rolesCollection.create((post) => {
+								post._raw = sanitizedRaw({ id: _id, description }, rolesCollection.schema);
+							});
+						});
+					}
+				}
+				if (/removed/.test(type)) {
+					const db = database.active;
+					const rolesCollection = db.get('roles');
+					try {
+						const rolesRecord = await rolesCollection.find(_id);
+						await db.action(async() => {
+							await rolesRecord.destroyPermanently();
+						});
+					} catch (err) {
+						console.log(err);
+					}
+				}
+				setRoles();
```

Comment:
When `'stream-roles'` is triggered, one role was changed.
We shouldn't overwrite the whole `roles` reducer because of it.
We should have extra redux actions to change a single role on the reducer.